litmus-agent-subscriber/src/main.go

```diff
@@ -0,0 +1,324 @@
+	} else {
+		cm, err = clientset.CoreV1().ConfigMaps(NAMESPACE).Update(context.TODO(), &configMap, metav1r.UpdateOptions{})
+		if err != nil {
+			utils.Red.Println("\n❌ Cannot update configmap " + configmapName + " : " + err.Error() + "\n")
+			os.Exit(1)
+		}
+
+	}
+	_ = cm
+}
+
+func createAgent(credentials types.Credentials) {
+	var newAgent types.Agent
+	newAgent.AgentName = AGENT_NAME
+	newAgent.Namespace = NAMESPACE
+	newAgent.Description = AGENT_DESCRIPTION
+	newAgent.ProjectId = LITMUS_PROJECT_ID
+	// Mode
+	// 1. cluster
+	// 2. namespace
+	newAgent.Mode = "cluster"
+
+	// -- OPTIONNAL -- //
+	newAgent.ClusterType = "external"
+	newAgent.NodeSelector = ""
+	// PlatformName
+	// 1. AWS
+	// 2. GKE
+	// 3. Openshift
+	// 4. Rancher
+	// 5. Others
+	newAgent.PlatformName = "Others"
+	newAgent.ServiceAccount = SERVICE_ACCOUNT_NAME
+	newAgent.SAExists = true
+	newAgent.NsExists = true
+
+	t := time.Now()
+	configMapData := make(map[string]string)
+	configMapData["SERVER_ADDR"] = LITMUS_BACKEND_URL + "/query"
+	configMapData["VERSION"] = APP_VERSION
+	configMapData["IS_CLUSTER_CONFIRMED"] = "false"
+	configMapData["START_TIME"] = t.Format("20060102150405")
+	test := `["app.kubernetes.io/instance=` + RELEASE_NAME + `"]`
+	configMapData["COMPONENTS"] = "DEPLOYMENTS: " + test
+	configMapData["AGENT_SCOPE"] = newAgent.Mode
+
+	var clusterID string
+	agentExist := GetAgentWithName(credentials, newAgent.AgentName)
+	if (agentExist == agentData{}) {
+		utils.White_B.Println("\n🚀 Registering new agent !! 🎉")
+		agent, err := apis.ConnectAgent(newAgent, credentials)
+		if err != nil {
+			utils.Red.Println("\n❌ Agent connection failed: " + err.Error() + "\n")
+			os.Exit(1)
+		}
+
+		// Print error message in case Data field is null in response
+		if (agent.Data == apis.AgentConnect{}) {
+			utils.PrintInJsonFormat(agent)
+			utils.Red.Println("\n❌ Agent connection failed, null response")
+			os.Exit(1)
+		}
+		clusterID = agent.Data.UserAgentReg.ClusterID
+		configMapData["CLUSTER_ID"] = clusterID
+		reqCluster := GetAgentWithID(credentials, agent.Data.UserAgentReg.ClusterID)
+		if (reqCluster == agentData{}) {
+			utils.Red.Println("\n❌ Agent Registered failed: " + err.Error() + "\n")
+			os.Exit(1)
+		}
+
+		// Checking if cluster with given clusterID and accesskey is present
+		configMapData["ACCESS_KEY"] = reqCluster.AccessKey
+
+		utils.White_B.Println("\n🚀 Agent Registered Successful!! 🎉")
+	} else {
+		clusterID = agentExist.ClusterID
+
+		utils.White_B.Println("\n🚀 Agent Already Registered!! 🎉")
+
+		configMapData["CLUSTER_ID"] = clusterID
+		configMapData["ACCESS_KEY"] = agentExist.AccessKey
+                configMapData["IS_CLUSTER_CONFIRMED"] = strconv.FormatBool(agentExist.IsClusterConfirmed)
+
+	}
+
+	createConfigMap(CONFIG_MAP_NAME, configMapData)
+
+	configMapWorkflowController := make(map[string]string)
+	configMapWorkflowController["config"] = `    containerRuntimeExecutor: k8sapi
+    executor:
+      imagePullPolicy: IfNotPresent
```

Comment:
should also be user-configurable not hardcoded, the runtime can differ depending on user